fineract-client/src/test/java/org/apache/fineract/integrationtests/newstyle/DocumentTest.java

```diff
@@ -0,0 +1,114 @@
+        assertThat(ok(fineract().documents.retreiveAllDocuments("clients", clientId))).isNotNull();
+    }
+
+    @Test
+    @Order(2)
+    void createDocument() throws IOException {
+        String name = "Test";
+        Part part = Parts.fromFile(testFile);
+        String description = null;
+        // TODO used var in tests when moved from fineract-client (Java 8 only) to new module
+        PostEntityTypeEntityIdDocumentsResponse response = ok(
+                fineract().documents.createDocument("clients", clientId, part, name, description));
+        assertThat(response.getResourceId()).isNotNull();
+        assertThat(response.getResourceIdentifier()).isNotEmpty();
+        documentId = response.getResourceId();
+    }
+
+    @Test
+    @Order(3)
+    void getDocument() throws IOException {
+        GetEntityTypeEntityIdDocumentsResponse doc = ok(fineract().documents.getDocument("clients", clientId, documentId));
+        assertThat(doc.getName()).isEqualTo("Test");
+        assertThat(doc.getFileName()).isEqualTo(testFile.getName());
+        assertThat(doc.getDescription()).isNull();
+        assertThat(doc.getId()).isEqualTo(documentId);
+        assertThat(doc.getParentEntityType()).isEqualTo("clients");
+        assertThat(doc.getParentEntityId()).isEqualTo(clientId);
+        // TODO huh?! It's more than uploaded file; seems like a bug - it's including create body, not just file size
+        assertThat(doc.getSize()).isEqualTo(testFile.length() + 385);
+        // TODO huh?! MIME is always text/plain instead of image/jpeg... :(
+        assertThat(doc.getType()).isEqualTo("text/plain");
+        // TODO doc.getStorageType() shouldn't be exposed by the API?!
+    }
+
+    @Test
+    @Order(4)
+    void downloadFile() throws IOException {
+        ResponseBody r = ok(fineract().documents.downloadFile("clients", clientId, documentId));
+        assertThat(r.contentType()).isEqualTo(MediaType.get("text/plain")); // TODO wrong, bug; needs to be "image/jpeg"
+                                                                            // (as above)
+        assertThat(r.bytes().length).isEqualTo(testFile.length());
+        // NOK: assertThat(r.contentLength()).isEqualTo(testFile.length());
+    }
+
+    @Test
+    @Order(10)
+    void updateDocument() throws IOException {
+        String newName = "Test changed name";
+        String newDescription = getClass().getName();
+        ok(fineract().documents.updateDocument("clients", clientId, documentId, null, newName, newDescription));
+
+        GetEntityTypeEntityIdDocumentsResponse doc = ok(fineract().documents.getDocument("clients", clientId, documentId));
+        assertThat(doc.getName()).isEqualTo(newName);
+        assertThat(doc.getDescription()).isEqualTo(newDescription);
+    }
+
+    @Test
+    @Order(99)
+    void deleteDocument() throws IOException {
+        ok(fineract().documents.deleteDocument("clients", clientId, documentId));
+    }
```

Comment:
Probably would make sense to check that get doesn't return the deleted document?

Reply:
done, added: `assertThat(fineract().documents.getDocument("clients", clientId, documentId)).hasHttpStatus(404);`